File: Linguagem C++/Matrizes/DIagonalNegativos.cpp
#include <iostream>

using namespace std;

int main()
{

    int n,cont;

    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int mat[n][n];

    for (int i=0;i<=n-1;i++){
        for (int j=0;j<=n-1;j++){
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        };
    };

    cout << "DIAGONAL PRINCIPAL:" << endl;

    for (int i=0;i<=n-1;i++){
        for (int j=0;j<=n-1;j++){
            if (i == j){
                cout << mat[i][j] << " ";
            };
        };
    };
    cout << endl;

    // Quantidade de negativos
    cont = 0;
    for (int i=0;i<=n-1;i++){
        for (int j=0;j<=n-1;j++){
            if (mat[i][j] < 0){
                cont++;
            };
        };
    };
    cout << "QUANTIDADE DE NEGATIVOS: " << cont << endl;


    return 0;
}

File: Linguagem C#/Matrizes/DiagonalNegativos.cs
using System;

namespace DiagonalNegativos {
    class Program {
        static void Main(string[] args) {

            int n,cont;

            Console.Write("Qual a ordem da matriz? ");
            n = int.Parse(Console.ReadLine());

            int[,] mat = new int[n, n];

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    Console.Write("Elemento ["+i+","+j+"]: ");
                    mat[i, j] = int.Parse(Console.ReadLine());
                };
            };

            Console.WriteLine("DIAGONAL PRINCIPAL:");
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    if (j == i) {
                        Console.Write(mat[i,j]+"  ");
                    };
                };
            };
            Console.WriteLine("");

            cont = 0;
            for (int i=0;i<n;i++) {
                for (int j = 0; j < n; j++) {
                    if (mat[i,j] < 0) {
                        cont++;
                    };
                };
            };
            Console.WriteLine("QUANTIDADE DE NEGATIVOS = "+cont);
        }
    }
}
